### PR TITLE
Nipd branch

### PR DESCRIPTION
**Pull request by erg**
All the stack shufflers you need but were afraid to ask for in core/. Taking opinions...

To bootstrap: http://downloads.factorcode.org/images/nipd/
``./factor -i=boot.unix-x86.64.image``

### CODE REVIEW

File: extra/xkcd/xkcd.factor [hunk 1]
```diff
@@ -13,8 +13,8 @@ IN: xkcd
 
 : comic-image ( url -- image )
     http-get nip
-    R/ http:\\/\\/imgs\.xkcd\.com\\/comics\\/[^\.]+\.(png|jpg)/
-    first-match >string load-http-image ;
+    R/ \/\/imgs\.xkcd\.com\\/comics\\/[^\.]+\.(png|jpg)/
+    first-match >string "http:" prepend load-http-image ;
```
Comment (mrjbq7) [inline extra/xkcd/xkcd.factor]: ?

Comment (erg) [reply extra/xkcd/xkcd.factor]: It broke. This should go into master/ anyway, oops

**Comment by AlexIljin** [comment]
I like this branch : ))

**Comment by jonenst** [comment]
You could also replace the "curry keep" usages with the new keepd ?
```
$ grep "curry keep" -ri core/ basis/ extra/
core/io/io.factor:    [ 2dup (new-sequence-for-stream) swap ] dip curry keep
core/io/io.factor:    [ swap (new-sequence-for-stream) ] curry keep
core/io/io.factor:    [ [ stream-read-unsafe ] curry keep resize ] bi ; inline
core/io/streams/sequence/sequence.factor:    ] dip [ <sequence-copy> (copy) drop ] 3curry keep ; inline
core/sequences/sequences.factor:    swapd [ curry keep ] curry ; inline
basis/classes/struct/struct.factor:    [ [ (writer-quot) call( value struct -- ) ] with 2each ] curry keep ;
basis/classes/struct/struct.factor:            over [ swapd [ call( value struct -- ) ] curry keep ] [ 2drop ] if
basis/formatting/formatting.factor:    ] curry keep neg? [ CHAR: - prefix ] when ;
basis/formatting/formatting.factor:    ] curry keep neg? [ CHAR: - prefix ] when ;
extra/noise/noise.factor:    [ [ v> -1.0 float-4-with vand ] curry keep v+ ] bi ; inline
```

**Comment by erg** [comment]
@jonenst Good call, missed that one

**Comment by erg** [comment]
Merged in 9f213f96f6589534d2ed5b3875ca05386b9afd5e

**Pull request closed without merging**